uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/shared/components/leave-calendar.component.js

```diff
@@ -0,0 +1,351 @@
+/* eslint-env amd */
+
+define([
+  'common/angular',
+  'common/lodash',
+  'common/moment',
+  'leave-absences/shared/modules/components',
+  'leave-absences/shared/controllers/sub-controllers/leave-calendar-admin.controller',
+  'leave-absences/shared/controllers/sub-controllers/leave-calendar-manager.controller',
+  'leave-absences/shared/controllers/sub-controllers/leave-calendar-staff.controller'
+], function (angular, _, moment, components) {
+  components.component('leaveCalendar', {
+    bindings: {
+      contactId: '<',
+      roleOverride: '@?'
+    },
+    templateUrl: ['shared-settings', function (sharedSettings) {
+      return sharedSettings.sharedPathTpl + 'components/leave-calendar.html';
+    }],
+    controllerAs: 'calendar',
+    controller: ['$controller', '$q', '$log', '$rootScope', '$timeout',
+      'shared-settings', 'AbsencePeriod', 'AbsenceType', 'LeaveRequest',
+      'PublicHoliday', 'OptionGroup', 'Calendar', 'checkPermissions',
+      controller]
+  });
+
+  function controller ($controller, $q, $log, $rootScope, $timeout, sharedSettings, AbsencePeriod, AbsenceType, LeaveRequest, PublicHoliday, OptionGroup, Calendar, checkPermissions) {
+    $log.debug('Component: leave-calendar');
+
+    var subController, userRole;
+    var vm = this;
+
+    vm.absencePeriods = [];
+    vm.contacts = [];
+    vm.injectMonths = false;
+    vm.months = [];
+    vm.selectedMonths = null;
+    vm.selectedPeriod = null;
+    vm.showContactName = false;
+    vm.showFilters = false;
+    vm.supportData = {};
+    vm.loading = { calendar: true, page: true };
+    vm.filters = {
+      optionValues: {},
+      userSettings: {
+        contact: null,
+        contacts_with_leaves: false,
+        department: null,
+        level_type: null,
+        location: null,
+        region: null
+      }
+    };
+
+    vm.labelPeriod = labelPeriod;
+    vm.refresh = refresh;
+
+    (function init () {
+      setUserRole()
+      .then(initWatchers)
+      .then(injectSubController)
+      .then(function () {
+        return $q.all([
+          loadContacts(),
+          loadAbsencePeriods(),
+          loadSupportData()
+        ]);
+      })
+      .then(function () {
+        return vm.showFilters ? loadFiltersOptionValues() : _.noop;
+      })
+      .then(function () {
+        injectAndShowMonths();
+      })
+      .then(function () {
+        vm.loading.page = false;
+      });
+    }());
+
+    /**
+     * Creates a list of all the months in the currently selected period
+     */
+    function buildPeriodMonthsList () {
```

Comment:
Actually there was a different flaw here: if a period started on 20/3/2017 and ended on 2/4/2017, the months list would have been only "March" instead of "March, April"

I fixed this now